homeassistant/util/async_.py

```diff
@@ -171,7 +171,7 @@ def protected_loop_func(*args: _P.args, **kwargs: _P.kwargs) -> _R:
     return protected_loop_func
 
 
-async def gather_with_concurrency(
```

Comment:
Its possible a custom component is also using this. Maybe make `gather_with_concurrency = gather_with_limited_concurrency` below and remove in 6 moths?